diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/SortRuleWarnTest.kt

```diff
@@ -0,0 +1,223 @@
+package org.cqfn.diktat.ruleset.chapter3
+
+import com.pinterest.ktlint.core.LintError
+import org.cqfn.diktat.ruleset.constants.Warnings.WRONG_DECLARATIONS_ORDER
+import generated.WarningNames
+import org.cqfn.diktat.ruleset.rules.DIKTAT_RULE_SET_ID
+import org.cqfn.diktat.ruleset.rules.SortRule
+import org.cqfn.diktat.util.lintMethod
+import org.junit.jupiter.api.Tag
+import org.junit.jupiter.api.Test
+
+class SortRuleWarnTest {
+
+    private val ruleId = "$DIKTAT_RULE_SET_ID:sort-rule"
+
+    @Test
+    @Tag(WarningNames.WRONG_DECLARATIONS_ORDER)
+    fun `check simple correct enum`() {
+        lintMethod(SortRule(),
+                """
+                    |enum class Alph {
+                    |   A,
+                    |   B,
+                    |   C,
+                    |   D,
+                    |   ;
+                    |}
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.WRONG_DECLARATIONS_ORDER)
+    fun `check simple wrong enum`() {
+        lintMethod(SortRule(),
+                """
+                    |enum class Alph {
+                    |   D,
+                    |   C,
+                    |   A,
+                    |   B,
+                    |   ;
+                    |}
+                """.trimMargin(),
+                LintError(1, 17, ruleId, "${WRONG_DECLARATIONS_ORDER.warnText()} enum entries order is incorrect", true)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.WRONG_DECLARATIONS_ORDER)
+    fun `check correct enum`() {
+        lintMethod(SortRule(),
+                """
+                    |enum class ENUM {
+                    |   BLUE(0x0000FF),
+                    |   GREEN(0x00FF00),
+                    |   RED(0xFF0000),
+                    |   ;
+                    |}
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.WRONG_DECLARATIONS_ORDER)
+    fun `check wrong enum without semicolon`() {
+        lintMethod(SortRule(),
+                """
+                    |enum class ENUM {
+                    |   GREEN(0x00FF00),
+                    |   RED(0xFF0000),
+                    |   BLUE(0x0000FF),
+                    |}
+                """.trimMargin(),
+                LintError(1, 17, ruleId, "${WRONG_DECLARATIONS_ORDER.warnText()} enum entries order is incorrect", true)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.WRONG_DECLARATIONS_ORDER)
+    fun `check wrong enum without semicolon and last comma`() {
```

Comment:
Presence of semicolon and comma is more important in fixers, it doesn't really change logic of check.